packages/scaffold-config/test/unit/detect.spec.ts

```diff
@@ -265,162 +265,219 @@ describe('detectFramework', () => {
-    const projectRoot = await scaffoldMigrationProject('migration-ts-files-only')
+      for (const f of files) {
+        if (!filename) {
+          fs.rmSync(f)
+        } else if (!f.includes(filename)) {
+          fs.rmSync(f)
+        }
+      }
+    }
 
-    fakeDepsInNodeModules(projectRoot, [{ devDependency: 'typescript', version: '4.3.6' }])
+    it('with only .d.ts files', async () => {
+      const projectRoot = await scaffoldMigrationProject('migration-dts-files-only')
 
-    // detected based on `integration/**/*.tsx
-    removeAllTsFilesExcept(projectRoot, 'integration')
+      fakeDepsInNodeModules(projectRoot, [{ devDependency: 'typescript', version: '4.3.6' }])
 
-    const actual = detectLanguage({ projectRoot, pkgJson: {} as PkgJson })
+      const actual = detectLanguage({ projectRoot, pkgJson: {} as PkgJson, isMigrating: true })
 
-    expect(actual).to.eq('ts')
-  })
+      expect(actual).to.eq('js')
+    })
 
-  it('cypress.json project with a TypeScript commponent spec', async () => {
-    const projectRoot = await scaffoldMigrationProject('migration-ts-files-only')
+    it('with a TypeScript supportFile', async () => {
+      const projectRoot = await scaffoldMigrationProject('migration-ts-files-only')
 
-    fakeDepsInNodeModules(projectRoot, [{ devDependency: 'typescript', version: '4.3.6' }])
+      fakeDepsInNodeModules(projectRoot, [{ devDependency: 'typescript', version: '4.3.6' }])
 
-    // detected based on `integration/**/*.tsx
-    removeAllTsFilesExcept(projectRoot, 'component')
+      removeAllTsFilesExcept(projectRoot, 'support')
 
-    const actual = detectLanguage({ projectRoot, pkgJson: {} as PkgJson })
+      const actual = detectLanguage({ projectRoot, pkgJson: {} as PkgJson })
 
-    expect(actual).to.eq('ts')
-  })
+      expect(actual).to.eq('ts')
+    })
 
-  it('ignores node_modules when checking for tsconfig.json', async () => {
-    const projectRoot = await scaffoldMigrationProject('pristine-cjs-project')
+    it('with a TypeScript pluginsFile', async () => {
+      const projectRoot = await scaffoldMigrationProject('migration-ts-files-only')
 
-    fakeDepsInNodeModules(projectRoot, [{ devDependency: 'typescript', version: '4.3.6' }])
+      fakeDepsInNodeModules(projectRoot, [{ devDependency: 'typescript', version: '4.3.6' }])
 
-    await fs.mkdirp(path.join(projectRoot, 'node_modules', 'some-node-module'))
-    await fs.writeFile(path.join(projectRoot, 'node_modules', 'some-node-module', 'tsconfig.json'), '')
-    const pkgJson = fs.readJsonSync(path.join(projectRoot, 'package.json'))
+      removeAllTsFilesExcept(projectRoot, 'plugins')
 
-    const actual = detectLanguage({ projectRoot, pkgJson })
+      const actual = detectLanguage({ projectRoot, pkgJson: {} as PkgJson })
 
-    expect(actual).to.eq('js')
+      expect(actual).to.eq('ts')
+    })
+
+    it('cwith a TypeScript integration specs', async () => {
```

Comment:
Is this test the same as the one below it? I don't see a difference between the two

Reply:
Good catch - appears to have been a duplicate, just a slightly different name. Removed